plugin/trino-kafka/pom.xml

```diff
@@ -138,6 +138,11 @@
             <artifactId>kafka-clients</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.pcollections</groupId>
```

Comment:
Why we are using this library? Why not simple HashSet? Is it about we are not removing items from it when we are closing the stack?

Is it possible to have a test for a case where using this library is needed (closing the recursion)? Is it possible (and reasonable) to not use this dependency?

Reply:
please see my explanation https://github.com/trinodb/trino/pull/15312#discussion_r1043252444